cibuildwheel/__main__.py

```diff
@@ -120,15 +124,21 @@ def main() -> None:
                   file=sys.stderr)
             exit(2)
 
+    if platform not in PLATFORMS:
+        print(f'cibuildwheel: Unsupported platform: {platform}', file=sys.stderr)
+        exit(2)
+
     package_dir = Path(args.package_dir)
     output_dir = Path(args.output_dir)
 
     if platform == 'linux':
         repair_command_default = 'auditwheel repair -w {dest_dir} {wheel}'
     elif platform == 'macos':
         repair_command_default = 'delocate-listdeps {wheel} && delocate-wheel --require-archs x86_64 -w {dest_dir} {wheel}'
-    else:
+    elif platform == 'windows':
         repair_command_default = ''
+    else:
+        assert_never(platform)
```

Comment:
I suppose it's there to satisfy mypy(?), but `assert_never(platform)` isn't very descriptive, and requires a trip to the definition to see what it does... could it be `raise UnhandledValueError(platform)` or even `raise ValueError(platform)`?

Reply:
This is [a standard name](https://hakibenita.com/python-mypy-exhaustive-checking#the-future), and it's for exhaustiveness checking. If you forget to include all possible platforms, mypy can tell (via type narrowing), that `assert_never` can be reached, and will error out.

It hasn't made it into mypy since the "correct" solution would be pattern matching. Which last I checked was a complete mess. To me it seems like a simple form (only `_` matches, always use `:=` to save) would have been far simpler than that mess they are making now (adding lots of new, only-in-pattern-matching syntax).

Reply:
Since it's in `typing`, I don't really mind using `assert_never` (though it's arguably slightly weirdly named; I'd expect `assert_unreachable` or `unreachable` or so).

Reply:
Ahhhh, it checks that the condition is unreachable at 'compile' time. That's nice.

Not a great name, though. Not all that widely used, either: https://github.com/search?l=Python&q=assert_never&type=Code . I also saw reference to `assert_exhaustive(variable)`, which I quite like. I'm happy either way - up to you, @henryiii

Reply:
It's in PyRight's tests: https://github.com/microsoft/pyright/search?q=assert_never , and 10x more common than assert_exhaustive, so I'd rather go with this and hope there's a chance someone has seen it (can link to the blog post at the definition, if you'd like?), but I'd be perfectly happy with assert_exhaustive if you do have a strong preference.

Reply:
PS: It's not in typing, though, as far as I know.

Reply:
Okay, let's go with that.

Reply:
It's now in typing, with this name. :) https://github.com/python/cpython/pull/30842

Reply:
:) ohhh, it's assert_never because of the type `Never`. That makes a bit more sense.